File: HGSocketManager/Message.cs
using System.Text.Json.Serialization;

namespace HGSocketManager;

/// <summary>
/// A message from the socket.
/// </summary>
[Serializable]
public class Message
{
    /// <summary>
    /// The type of message.
    /// </summary>
    [JsonPropertyName("type")]
    public required string MessageType { get; set; }

    /// <summary>
    /// The content of the message.
    /// </summary>
    [JsonPropertyName("content")]
    public string? Content { get; set; }
}

File: client/src/lib/networking/message.ts
/**
 * Message interface
 * @type {string} type - The type of the message
 * @type {string} content - The content of the message
 */
export class Message {
  public type: string;
  public content: string;

  constructor(type: string, content: any) {
    this.type = type;
    this.content = JSON.stringify(content);
  }
}
